pkg/skaffold/deploy/helm/helm.go

```diff
@@ -295,8 +307,33 @@ func (h *Deployer) Deploy(ctx context.Context, out io.Writer, builds []graph.Art
 				return helm.UserErr(fmt.Sprintf("cannot expand chart path %q", r.ChartPath), err)
 			}
 
+			for _, dependency := range r.DependsOn {
```

Comment:
what about nested dependencies?  
ServiceA depends on ServiceB and ServiceB depends on Service C

Reply:
I have not created a test for this case but, in theory, it should work the same way. Here's why:

* The first action is to check all services that has dependents, then create the channels to use as wait controllers;
* So, In the case you're suggesting:
  * If ServiceC starts first, it will hit the signal break with channels; It's gonna wait for Service B to continue;
  * If ServiceB starts in second, it will hit the signal break too, and wait for ServiceA
  * ServiceA does not depends on anyone, so it deploys and send the "all clear" signal to his dependents
  * ServiceB get's the "all clear" signal, and starts this deployment. When finished, it will also sends his signal for ServiceC
  * Finally, ServiceC get's his all clear and do this deployment.

One scenario that **may** happen is a deadlock. I did not think about it on this PR. 

Another thing that I'm thinking is that this deserve a integration testing. I'm still learning about this repo and I will create a separated PR for that. This PR already has the label "size/L" 😱

Reply:
I suggest you to check this approach https://github.com/GoogleContainerTools/skaffold/pull/9588#issuecomment-2505671935, because the current approach will lead to deadlock(concurrency = 3 and 3 dependencies are waiting, for example).    
don't worry about the "size/L", it's ok)

Reply:
Ok thanks for the feedback @idsulik . I'll work on that. Now I see the point, it's easy to get to a deadlock situation because of the lack of resources.